fieldmanager.php

```diff
@@ -435,6 +440,93 @@ function fm_trigger_context_action() {
 }
 add_action( 'init', 'fm_trigger_context_action', 99 );
 
+/**
+ * Fire an action for the current Fieldmanager context for the REST API.
+ *
+ * This is separate from fm_trigger_context_action since rest_pre_dispatch fires on parse_request
+ * and is too late to be part of the existing context hooks.
+ *
+ * @param mixed           $result  Response to replace the requested version with. Can be anything
+ *                                 a normal endpoint can return, or null to not hijack the request.
+ * @param WP_REST_Server  $server  Server instance.
+ * @param WP_REST_Request $request Request used to generate the response.
+ */
+function fm_trigger_rest_context_action( $result, $server, $request ) {
+	// Get post type and taxonomy endpoints available for the REST API.
+	$post_type_endpoints = fm_get_registered_object_rest_base( 'post_types' );
+	$taxonomy_endpoints = fm_get_registered_object_rest_base( 'taxonomies' );
+
+	// Get the route for the request for comparison.
+	$route = $request->get_route();
+
+	// If the route is empty, we cannot continue.
+	if ( empty( $route ) ) {
+		return;
+	}
+
+	// Store matches to use with the context action.
+	$matches = array();
+
+	// Use regexes to find the right context.
+	if (
+		! empty( $post_type_endpoints )
+		&& preg_match( '/\/wp\/v2\/(' . implode( '|', array_keys( $post_type_endpoints ) ) . ')(\/?)(.*?)/', $route, $matches )
+	) {
+		// Post Types.
+		do_action( "fm_post_{$post_type_endpoints[ $matches[1] ]}" );
```

Comment:
Bump on this question.

Reply:
> Should this also fire fm_post? (Similarly, should fm_term be fired below?)

I am not sure, you might have more incite then me on this one.  However, based on your comment below about [leveraging the existing trigger functions](https://github.com/alleyinteractive/wordpress-fieldmanager/pull/583#discussion_r153648203) I think it might be best to try and incorporate the REST API requests into the trigger functions.